diff --git a/dom/indexedDB/moz.build b/dom/indexedDB/moz.build
--- a/dom/indexedDB/moz.build
+++ b/dom/indexedDB/moz.build
@@ -100,6 +100,7 @@ if CONFIG['CC_TYPE'] in ('clang', 'gcc')
     # Suppress gcc warning about a comparison being always false due to the
     # range of the data type
     SOURCES['Key.cpp'].flags += ['-Wno-error=type-limits']
+    SOURCES['ActorsParent.cpp'].flags += ['-fno-lto']
     CXXFLAGS += ['-Wno-error=shadow']
 
 LOCAL_INCLUDES += [
