# HG changeset patch
# Parent  4fae89dde1a0e5217f5d33fbd2246004e6597843

Index: mozilla-esr153/dom/media/platforms/ffmpeg/FFmpegRuntimeLinker.cpp
===================================================================
--- mozilla-esr153.orig/dom/media/platforms/ffmpeg/FFmpegRuntimeLinker.cpp
+++ mozilla-esr153/dom/media/platforms/ffmpeg/FFmpegRuntimeLinker.cpp
@@ -55,6 +55,8 @@ static const char* sLibs[] = {
   "libavcodec.so.60",
   "libavcodec.so.59",
   "libavcodec.so.58",
+  "libavcodec.so.58.134",
+  "libavcodec.so.58.91",
   "libavcodec-ffmpeg.so.58",
   "libavcodec-ffmpeg.so.57",
   "libavcodec-ffmpeg.so.56",
