# HG changeset patch
# Parent  3802b7b2b99885f3ad4d2ec3e2c157e38d04a168

Index: firefox-102.0/dom/media/platforms/ffmpeg/FFmpegRuntimeLinker.cpp
===================================================================
--- firefox-102.0.orig/dom/media/platforms/ffmpeg/FFmpegRuntimeLinker.cpp
+++ firefox-102.0/dom/media/platforms/ffmpeg/FFmpegRuntimeLinker.cpp
@@ -40,6 +40,8 @@ static const char* sLibs[] = {
 #else
   "libavcodec.so.59",
   "libavcodec.so.58",
+  "libavcodec.so.58.134",
+  "libavcodec.so.58.91",
   "libavcodec-ffmpeg.so.58",
   "libavcodec-ffmpeg.so.57",
   "libavcodec-ffmpeg.so.56",
