Index: firefox-128.0/media/libjpeg/jconfig.h
===================================================================
--- firefox-128.0.orig/media/libjpeg/jconfig.h
+++ firefox-128.0/media/libjpeg/jconfig.h
@@ -19,7 +19,7 @@
 #define MEM_SRCDST_SUPPORTED 1
 
 /* Use accelerated SIMD routines. */
-#if defined(__sparc__) || defined(__loongarch__)
+#if defined(__sparc__) || defined(__loongarch__) || defined(__s390__) || defined(__s390x__)
 #undef  WITH_SIMD
 #else
 #define WITH_SIMD 1
Index: firefox-128.0/media/libjpeg/jconfigint.h
===================================================================
--- firefox-128.0.orig/media/libjpeg/jconfigint.h
+++ firefox-128.0/media/libjpeg/jconfigint.h
@@ -81,7 +81,7 @@
 /* #undef D_ARITH_CODING_SUPPORTED */
 
 /* Use accelerated SIMD routines. */
-#if defined(__sparc__) || defined(__loongarch__)
+#if defined(__sparc__) || defined(__loongarch__) || defined(__s390__) || defined(__s390x__)
 #undef  WITH_SIMD
 #else
 #define WITH_SIMD 1
